plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/ConcurrentIndex.java

```diff
@@ -111,8 +112,11 @@ public ResourceDescriptionsData setProjectIndex(String projectName, ResourceDesc
 			oldProjectIndex = project2Index.put(projectName, projectIndex);
 			newWorkspaceConfig = workspaceConfig;
 		}
-		notifyListeners(newWorkspaceConfig, ImmutableMap.of(projectName, projectIndex), ImmutableList.of(),
```

Comment:
as discussed: new empty projects will no longer appear in `changedDescriptions` parameter of listener.